### PR TITLE
Feature/initial implementation

### PR DESCRIPTION
**Pull request by martinichka**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,62 @@
 # Gateway Module Interface
+
+This library is for coding and decoding UART frames for the LG8271/LG9271 modules. It is written for the TTN LoRa gateway.
```
Comment (htdvisser) [inline README.md]: Write `It is written for [The Things Gateway](https://github.com/TheThingsProducts/gateway/)`

File: README.md [hunk 2]
```diff
@@ -1 +1,62 @@
...
+
+## Platform dependancy
```
Comment (htdvisser) [inline README.md]: Dependency (also below)

File: README.md [hunk 3]
```diff
@@ -1 +1,62 @@
...
+The platform dependancy is decoupled by a small set of functions to be implemented per platform.
+
+Mutex for the UART write action
+
+	typedef void (*gateway_module_interface_write_lock_t)(bool lock);
```
Comment (htdvisser) [inline README.md]: Use fenced code blocks with language/highlighting in Markdown. See https://help.github.com/articles/creating-and-highlighting-code-blocks/

**Comment by htdvisser** [comment]
I'll let @KrishnaIyer review the C code. 

Would it be a good idea to put some config for clang-format in the repo?

**Comment by KrishnaIyer** [comment]
@martinichka 
1. This is the same source code as `108-ttn-lg8271` of bit-bucket, isn't it? If all the review comments are closed there, then this would be just a copy-paste thing.
2. Could you rather create a makefile and remove the dependency with Eclipse? If I have time today, I'll create one for you.

**Comment by martinichka** [comment]
@KrishnaIyer 1. yes, 2. done (see commit 0b88c16)

**Review by KrishnaIyer — APPROVED** [review]
ok since we have the Makefile and the code is just taken over from bitbucket, I'm good with the PR.

**Pull request merged**